File: app/src/main/java/org/nitri/opentopo/model/TrackDetail.java
package org.nitri.opentopo.model;

import org.nitri.opentopo.domain.DistancePoint;

import java.util.Collections;
import java.util.List;

public class TrackDetail {

    private String name;
    private List<DistancePoint> distancePoints;

    public TrackDetail(String name, List<DistancePoint> distancePoints) {
        this.name = name;
        this.distancePoints = Collections.unmodifiableList(distancePoints);
    }

    public String getName() {
        return name;
    }

    public List<DistancePoint> getDistancePoints() {
        return distancePoints;
    }
}

File: app/src/main/java/org/nitri/opentopo/model/TrackDetail.kt
package org.nitri.opentopo.model

import org.nitri.opentopo.domain.DistancePoint
import java.util.Collections

class TrackDetail(val name: String, distancePoints: List<DistancePoint>?) {
    val distancePoints: List<DistancePoint> = TODO()

    init {
        if (distancePoints != null) {
            this.distancePoints = distancePoints
        }
    }
}
